lib/widgets/pages/settings/settings_page.dart

```diff
@@ -42,7 +44,28 @@ class SettingsPage extends StatelessWidget {
                       state.user.email,
                       style: AppTextStyle.settingValue,
                     ),
-                    onTap: () {},
+                    onTap: () {
+                      Navigator.push<void>(
+                        context,
+                        MaterialPageRoute(
+                          builder: (BuildContext context) =>
+                              AppScaffold.withTitle(
+                            title: Strings.changeEmail,
+                            body: EmailBody(
+                              initialValue: state.user.email,
+                              onSubmit: (context, email) {
+                                context.read<UserCubit>().setUserEmail(email);
+                                Navigator.pop(context);
+                                //FIXME: should we sign the user out after they changed their email? i think their token may become invalid
```

Comment:
Might be a good idea. I just tried to check in the backend, and it does seem that we are reliant on the email embedded in the token for a few things